core-java-modules/core-java-lang-math-4/src/test/java/com/baeldung/PercentageDifferenceBetweenTwoNumbersTest/PercentageDifferenceBetweenTwoNumbersTest.java

```diff
@@ -0,0 +1,84 @@
+package com.baeldung.PercentageDifferenceBetweenTwoNumbersTest;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.baeldung.math.percentagedifferecebetweentwonumbers.PercentageDifferenceBetweenTwoNumbers;
+import org.junit.jupiter.api.Test;
+
+public class PercentageDifferenceBetweenTwoNumbersTest {
+
+  @Test
+  void calculatePercentageDifferenceBetweenTwoNumbers() {
+    double V1 = 50.0;
+    double V2 = 70.0;
+    double expected = 33.33;
+    double result = PercentageDifferenceBetweenTwoNumbers.calculatePercentageDifference(V1, V2);
+    assertEquals(expected, result, 0.01, "Percentage difference should be correctly calculated.");
+  }
+
+  @Test
+  void whenV1AndV2AreEqual_thenPercentageDifferenceIsZero() {
+    double V1 = 50.0;
+    double V2 = 50.0;
+    double expected = 0.0;
+    double result = PercentageDifferenceBetweenTwoNumbers.calculatePercentageDifference(V1, V2);
+    assertEquals(
+        expected, result, "Percentage difference should be zero when both values are equal.");
+  }
+
+  @Test
+  void whenV1IsGreaterThanV2_thenCalculateCorrectPercentageDifference() {
+    double V1 = 70.0;
+    double V2 = 50.0;
+    double expected = 33.33;
+    double result = PercentageDifferenceBetweenTwoNumbers.calculatePercentageDifference(V1, V2);
+    assertEquals(expected, result, 0.01, "Percentage difference should be correctly calculated.");
+  }
+
+  @Test
+  void whenV2IsGreaterThanV1_thenCalculateCorrectPercentageDifference() {
+    double V1 = 50.0;
+    double V2 = 70.0;
+    double expected = 33.33;
+    double result = PercentageDifferenceBetweenTwoNumbers.calculatePercentageDifference(V1, V2);
+    assertEquals(expected, result, 0.01, "Percentage difference should be correctly calculated.");
+  }
+
+  @Test
+  void whenOneValueIsZero_thenCalculateCorrectPercentageDifference() {
+    double V1 = 0.0;
+    double V2 = 50.0;
+    double expected = 200.0;
+    double result = PercentageDifferenceBetweenTwoNumbers.calculatePercentageDifference(V1, V2);
+    assertEquals(
+        expected,
+        result,
+        "Percentage difference should be correctly calculated when one value is zero.");
+  }
+
+  @Test
+  void whenAverageIsZero_thenThrowIllegalArgumentException() {
+    double V1 = -50.0;
+    double V2 = 50.0;
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> {
```

Comment:
This can be simplified slightly into  

`() -> PercentageDifferenceBetweenTwoNumbers.calculatePercentageDifference(V1, V2)`